### DISCUSSION TITLE
Can the unique BLE UUID of the ESP chip be exposed via NimBLEServer?

### DISCUSSION THREAD

**Discussion by konacurrents** [h2zero/NimBLE-Arduino#418] [General]
I've been using the NimBLE code and have a large number of servers running the same service name and services/characteristics.

What I would like it to know the unique UUID that the actual ESP BLE device knows (which I understand is unique like a MAC address).

Those looking at advertising device can see that UUID, but I'm not able to get access to its value in the software. If I can get that value, I can equate it to a user supplied device name, and use that in my connection to specific named devices (versus the service name).

I believe this is a GATT requirement. 

Any help would be appreciates.
thanks.

**Comment by h2zero** [comment]
I'm not sure which UUID you are referring to? If it's the one you see on a phone app it is generated by the phone as a replacement for that MAC address (for privacy). The server device has no knowledge of this.

**Reply by konacurrents** [reply]
Yes the UUID I see on the phone (CBPeripherial). I just confirmed from 2 different phones seeing different UUID's for the same device. Darn..

So there is no way to get the BLE's MAC address, even if just internally? If so, it could decide (for privacy) whether to do something with this.

But that's interesting and helps me from digging deeper.

**Reply by konacurrents** [reply]
My plan now is to tack my device name onto the service name. Then I can find specific BLE servers based on their extensions. 

So my M5 BLE service name: `PTClicker` becomes `PTClicker:ScoobyDoo` but with same characteristics. 

My app looks for services that start with common name then parse out the device name, eg `ScoobyDoo`. 

(The ESP and M5 apps deal with a dog feeding capability. Look at PetTutor and iDogWatch.com)
